geode-core/src/main/java/org/apache/geode/internal/cache/BucketAdvisor.java

```diff
@@ -351,6 +351,23 @@ void removeBucket() {
     setHosting(false);
   }
 
+  /**
+   * Return all members hosting this bucket
+   *
+   * @return members hosting bucket
+   */
+  public Set<InternalDistributedMember> getAllHostingMembers() {
+    Set<InternalDistributedMember> recipients = new HashSet<>();
+    Profile[] locProfiles = profiles; // grab current profiles
+    for (Profile profile : locProfiles) {
+      BucketProfile bp = (BucketProfile) profile;
+      if (bp.isHosting) {
+        recipients.add(bp.getDistributedMember());
+      }
+    }
+    return recipients;
+  }
+
   /**
```

Comment:
I'm pretty sure `adviseInitialized` does the same thing as the new `getAllHostingMembers` method.

The main difference is that `adviseFilter` calls `initializationGate()`, but the `sendQueueRemovalMesssageForDroppedEvent` method used to call `getAllRecipients` which calls `adviseDataStore` which calls `adviseFilter`.

adviseInitialized calls adviseFilter with `bucketProfile.isHosting`:
```
Set<InternalDistributedMember> adviseInitialized() {
  return adviseFilter(profile -> {
    assert profile instanceof BucketProfile;
    BucketProfile bucketProfile = (BucketProfile) profile;
    return bucketProfile.isHosting;
  });
}
```
So I think you can use this in `ParallelGatewaySenderQueue.getAllRecipientsForEvent`:
```
Set<InternalDistributedMember> bucketMembers =
    bucketRegionQueue.getBucketAdvisor().adviseInitialized();
```